### PR TITLE
fix(core): export animation classes required for Renderer impl

### LINKED ISSUE #14001: Core API: implementing Renderer requires types from private exports
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
The `animate` abstract method from the `Renderer` abstract class is defined using types only available though private exports, namely: `AnimationStyles`, `AnimationKeyframe`, `AnimationPlayer`, `NoOpAnimationPlayer`.

**Expected behavior**
All the types to implement `animate` are public, or the `animate` is not required in the `Renderer`.

**Minimal reproduction of the problem with instructions**
1. Define a TypeScript Angular module.
2. In the module, implement and provide a `RootRenderer` injectable service with a custom `Renderer` implemention.
3. TypeScript compiler refuses to compile the `Renderer` without the `animate` method. After trying to add it, the issue occurs.

**What is the motivation / use case for changing the behavior?**
The `Renderer` is a public API, although experimental. Implementing a `Renderer` makes low-level DOM adjustments possible, those are sometimes essential to use together with third-party component libraries, such as `Polymer`.

As for now, a custom `Renderer` implementation is not achievable without importing the animation types mentioned above from private core exports.

**Please tell us about your environment:**

* **Angular version:** 2.3.X

* **Browser:** all

 
* **Language:** TypeScript 2.0.10

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Closes #14001

### CODE REVIEW

**Comment by mhevery** [comment]
Can you fix the build

**Comment by DzmitryShylovich** [comment]
sure working on it

**Comment by DzmitryShylovich** [comment]
@mhevery done

File: tools/public_api_guard/core/index.d.ts [hunk 1]
```diff
@@ -669,6 +686,24 @@ export declare class NgZone {
 /** @experimental */
 export declare const NO_ERRORS_SCHEMA: SchemaMetadata;
 
+/** @experimental */
+export declare class NoOpAnimationPlayer implements AnimationPlayer {
```
Comment (matsko) [inline tools/public_api_guard/core/index.d.ts]: Why do we need to export the `NoOpAnimationPlayer`?

Comment (DzmitryShylovich) [inline tools/public_api_guard/core/index.d.ts]: It's not necessary. Just for convenience so users don't need to create their own noop imp.

**Comment by mhevery** [comment]
Please remove as we want to keep the API surface minimal. Unless it has to
be there, it should not.

On Mon, Jan 23, 2017 at 3:23 PM, Dzmitry Shylovich <notifications@github.com

**Comment by DzmitryShylovich** [comment]
done

**Pull request merged**